src/rpcblockchain.cpp

```diff
@@ -1056,6 +1056,135 @@ UniValue rain(const UniValue& params, bool fHelp)
     return res;
 }
 
+UniValue rainbymagnitude(const UniValue& params, bool fHelp)
+{
+    if (fHelp || (params.size() < 1 || params.size() > 2))
+        throw runtime_error(
+                "rainbymagnitude <amount> [message]\n"
+                "\n"
+                "<amount> --> Required: Specify amount of coints in double to be rained\n"
+                "[message] -> Optional: Provide a message rained to all rainees\n"
+                "\n"
+                "rain coins by magnitude on network");
+
+    UniValue res(UniValue::VOBJ);
+
+    double dAmount = params[0].get_real();
+
+    if (dAmount <= 0)
+        throw runtime_error("Amount must be greater then 0");
+
+    std::string sMessage = "";
+
+    if (params.size() > 1)
+        sMessage = params[1].get_str();
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    CWalletTx wtx;
+    wtx.mapValue["comment"] = "Rain By Magnitude";
+    std::set<CBitcoinAddress> setAddress;
+    std::vector<std::pair<CScript, int64_t> > vecSend;
+
+    wtx.hashBoinc = "<NARR>Rain By Magnitude: " + MakeSafeMessage(sMessage) + "</NARR>";
+
+    // Gather Magnitude data
+    std::string sSuperblock = ReadCache("superblock", "magnitudes").value;
+
+    if (sSuperblock.empty())
+        throw runtime_error("Unable to load superblock data");
+
+    std::vector<std::string> vSuperblock = split(sSuperblock, ";");
+
+    double dTotalAmount = 0;
+    int64_t nTotalAmount = 0;
+
+    StructCPID structcpid = mvNetwork["NETWORK"];
+
+    // Use total network magnitude here as using 115000 can results in a lower then intended sent amount due to a missing project
+    double dTotalNetworkMagnitude = structcpid.NetworkMagnitude;
+
+    for (unsigned int i = 0; i < vSuperblock.size(); i++)
+    {
+        if (vSuperblock[i].length() > 1)
+        {
+            // Find GRC address in beacon
+            std::string sCPID = ExtractValue(vSuperblock[i], ",", 0);
+            double dMagnitude = RoundFromString(ExtractValue(vSuperblock[i], ",", 1), 0);
+
+            // No magnitude means no value to be rained on
+            if (dMagnitude == 0)
+                continue;
+
+            std::string scacheContract = ReadCache("beacon", sCPID).value;
+
+            // Should never occur but we know seg faults can occur in some cases
+            if (scacheContract.empty())
+                continue;
+
+            std::string sContract = DecodeBase64(scacheContract);
+            std::string sGRCAddress = ExtractValue(sContract, ";", 2);
+
+            double dPayout = Round(((dMagnitude / dTotalNetworkMagnitude) * dAmount), 4);
```

Comment:
Skip this recipient if the payment would be 0?

This will also cause the total payment to become less than the specified.

Reply:
I like the idea of no rounding. and yes 0 magnitude should be ignored.

Reply:
gonna eliminate the round and also if they have 0 mag its already skipped